src/Microsoft.Tye.Core/ProjectReader.cs

```diff
@@ -166,18 +174,45 @@ private static void EvaluateProject(OutputContext output, ProjectServiceBuilder
                 throw new CommandException($"Failed to load project: '{project.ProjectFile.FullName}'.", ex);
             }
 
-            // Currently we only log at debug level.
-            var logger = new ConsoleLogger(
-                verbosity: LoggerVerbosity.Normal,
-                write: message => output.WriteDebug(message),
-                colorSet: null,
-                colorReset: null);
-
             try
             {
                 AssemblyLoadContext.Default.Resolving += ResolveAssembly;
+
+                output.WriteDebugLine($"Restoring project '{project.ProjectFile.FullName}'.");
+
+                // Similar to what MSBuild does for restore:
+                // https://github.com/microsoft/msbuild/blob/3453beee039fb6f5ccc54ac783ebeced31fec472/src/MSBuild/XMake.cs#L1417
+                //
+                // We need to do restore as a separate operation
+                var restoreRequest = new BuildRequestData(
+                    projectInstance,
+                    targetsToBuild: new[] { "Restore" },
+                    hostServices: null,
+                    flags: BuildRequestDataFlags.ClearCachesAfterBuild | BuildRequestDataFlags.SkipNonexistentTargets | BuildRequestDataFlags.IgnoreMissingEmptyAndInvalidImports);
+
+                var parameters = new BuildParameters(projectCollection)
+                {
+                    Loggers = new[] { logger, },
+                };
+
+                // We don't really look at the result, because it's not clear we should halt totally
```

Comment:
This is worth a `REVIEW:` comment

Reply:
what is a review comment?

Reply:
A comment which collects dust 😆

Reply:
ok, then, I'm deleting the comment in advance :)